remotes/docker/pusher.go

```diff
@@ -193,7 +193,9 @@ func (p dockerPusher) push(ctx context.Context, desc ocispec.Descriptor, ref str
 
```

Comment:
Wondering if instead, we could add a `defer` here;

```suggestion
		defer resp.Body.Close()
```

Reply:
Should the `default` case below also close? https://github.com/containerd/containerd/blob/f0a32c66dad1e9de716c9960af806105d691cd78/remotes/docker/pusher.go#L206-L209

Reply:
Thanks, moved it out of the switch and added the defer.  I was thinking that `NewUnexpectedStatusErr` closed the body, but it does not.